ltiauthenticator/__init__.py

```diff
@@ -131,9 +131,15 @@ def authenticate(self, handler, data=None):
         for k, values in handler.request.body_arguments.items():
             args[k] = values[0].decode() if len(values) == 1 else [v.decode() for v in values]
 
+        if 'x-forwarded-proto' in handler.request.headers and "https" in handler.request.headers['x-forwarded-proto']:
```

Comment:
I did it this way because of the value of `x-forwarded-proto` I have.  
It is equal to `X-Forwarded-Proto: https,http` but I don't know why and it doesn't seem to be a valid value. So I'll do it your way and try to fix my nginx config

Reply:
Thanks! yeah, https,http does not look like a valid x-forwarded-proto value to me! LMK if you get that under control :)

Reply:
Ok so it was the `configurable-http-proxy` that append `http` to the header afterwards but you can disable it using : 
```python
# Disable overwriting of x-forwarded-proto
c.JupyterHub.proxy_cmd = ['configurable-http-proxy', '--no-x-forward', '--log-level=debug']
```

Reply:
Commas are allowed in X-Forwarded-Proto, which indicate the protocol of multiple proxy hops. If you want the 'outermost' protocol, that would be:

```python
proto = handler.request.headers['x-forwarded-proto'].split(',')[0]
```

Reply:
Tornado has the [same logic](https://github.com/tornadoweb/tornado/blob/v5.0.1/tornado/httpserver.py#L292) but with the conservative default of only trusting the innermost proxy.